### ISSUE TITLE
check_spelling errored on a verbatim chunk in md file

### ISSUE THREAD

**Issue by hansvancalster** [inbo/checklist#192]
This definitively is an esotheric edge case. I had a `CONTRIBUTING.md` file that contained an example R code chunk example formatted as follows (simplified - and yes it's important that two times stringsAsFactors occurred):

````
<some text>
```
  df1 <- data.frame(
     stringsAsFactors = FALSE,
     a  = "a"
  )
  df2 <- data.frame(
      stringsAsFactors = FALSE,
      b = "b"
  )
```
<some more text>
````

This tripped up `checklist::check_spelling` resulting in a hard to debug error. 
The error could be "fixed" from my side by removing all `stringsAsFactors = FALSE` occurrences since this is the default since R 4.0.0. 
The below reprex , made with help of gemini, suggests a possible bug fix. I think the fix is useful on its own because it avoids the error, but it would be more useful if checklist completely ignored the above "verbatim" text (code) between the triple backticks since this also resulted in (after the error was fixed) false positive spelling mistakes (i.e. any other R code function names that are not in the dictionary). So these are actually two issues: one bug fix, and one enhancement.

``` r
# 1. Simulate the exact nested list structure checklist uses.
# The outer list (representing files) is unnamed, and the word 
# appears multiple times.
result <- list(
  list(
    "stringsAsFactors" = data.frame(line = 10, message = "stringsAsFactors"),
    "stringsAsFactors" = data.frame(line = 14, message = "stringsAsFactors")
  )
)

# 2. checklist runs unlist(recursive = FALSE). 
# Because the outer list is unnamed, the inner names remain exactly "stringsAsFactors".
unlisted_result <- unlist(result, recursive = FALSE)

# 3. do.call() passes these to rbind(), triggering the exact bug:
do.call(rbind, unlisted_result)
#> Error in `rbind()`:
#> ! formal argument "stringsAsFactors" matched by multiple actual arguments

# 4. The proposed fix: strip the names before passing to do.call
do.call(rbind, unname(unlisted_result))
#>   line          message
#> 1   10 stringsAsFactors
#> 2   14 stringsAsFactors
```

<sup>Created on 2026-03-27 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

<details style="margin-bottom:10px;">

<summary>

Session info
</summary>

``` r
sessioninfo::session_info()
#> Warning in system2("quarto", "-V", stdout = TRUE, env = paste0("TMPDIR=", :
#> running command '"quarto"
#> TMPDIR=C:/Users/hans_vancalster/AppData/Local/Temp/RtmpemT3Hw/file30b86be563b2
#> -V' had status 1
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.5.1 (2025-06-13 ucrt)
#>  os       Windows 11 x64 (build 26200)
#>  system   x86_64, mingw32
#>  ui       RTerm
#>  language (EN)
#>  collate  Dutch_Belgium.utf8
#>  ctype    Dutch_Belgium.utf8
#>  tz       Europe/Brussels
#>  date     2026-03-27
#>  pandoc   3.4 @ C:/Program Files/RStudio/resources/app/bin/quarto/bin/tools/ (via rmarkdown)
#>  quarto   NA @ C:\\PROGRA~1\\RStudio\\RESOUR~1\\app\\bin\\quarto\\bin\\quarto.exe
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version date (UTC) lib source
#>  cli           3.6.5   2025-04-23 [1] CRAN (R 4.5.0)
#>  digest        0.6.39  2025-11-19 [1] CRAN (R 4.5.2)
#>  evaluate      1.0.5   2025-08-27 [1] CRAN (R 4.5.2)
#>  fastmap       1.2.0   2024-05-15 [1] CRAN (R 4.5.0)
#>  fs            1.6.6   2025-04-12 [1] CRAN (R 4.5.0)
#>  glue          1.8.0   2024-09-30 [1] CRAN (R 4.5.0)
#>  htmltools     0.5.9   2025-12-04 [1] CRAN (R 4.5.2)
#>  knitr         1.51    2025-12-20 [1] CRAN (R 4.5.2)
#>  lifecycle     1.0.5   2026-01-08 [1] CRAN (R 4.5.2)
#>  otel          0.2.0   2025-08-29 [1] CRAN (R 4.5.2)
#>  pillar        1.11.1  2025-09-17 [1] CRAN (R 4.5.2)
#>  reprex        2.1.1   2024-07-06 [1] CRAN (R 4.5.0)
#>  rlang         1.1.7   2026-01-09 [1] CRAN (R 4.5.2)
#>  rmarkdown     2.30    2025-09-28 [1] CRAN (R 4.5.2)
#>  rstudioapi    0.18.0  2026-01-16 [1] CRAN (R 4.5.2)
#>  sessioninfo   1.2.3   2025-02-05 [1] CRAN (R 4.5.0)
#>  vctrs         0.7.1   2026-01-23 [1] CRAN (R 4.5.2)
#>  withr         3.0.2   2024-10-28 [1] CRAN (R 4.5.0)
#>  xfun          0.56    2026-01-18 [1] CRAN (R 4.5.2)
#>  yaml          2.3.12  2025-12-10 [1] CRAN (R 4.5.2)
#> 
#>  [1] C:/R/library
#>  [2] C:/R/R-4.5.1/library
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

</details>

**Comment by ThierryO** [comment]
This chunck should have been [excluded](https://github.com/inbo/checklist/blob/e61eb5df8253db3a84f56bda2f9147b66c6faf21/R/spelling_parse.R#L175) automatically.

**Comment by ThierryO** [comment]
Are you sure you have matching chunk delimiters?

**Comment by hansvancalster** [comment]
It appears that I had two files `CONTRIBUTING.md` one that was gitignored inside docs/ folder. In that file backticks were missing completely. Another one in the .github/ folder with correct backticks. So perhaps I saw this happening with the former. At some point during debugging I excluded docs/ folder from spell checking (shouldn't gitignored folders be automatically ignored by the spell checker too?).
